cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -328,3 +329,28 @@ func ValidateAPIEndpoint(c *kubeadm.MasterConfiguration, fldPath *field.Path) fi
 	}
 	return allErrs
 }
+
+// ValidateIgnoreChecksErrors validates duplicates in ignore-checks-errors flag.
+func ValidateIgnoreChecksErrors(ignoreChecksErrors []string, skipPreflightChecks bool) (sets.String, error) {
+	ignoreErrors := sets.NewString()
+	allErrs := field.ErrorList{}
+
+	// TODO: remove once deprecated flag --skip-preflight-checks is removed.
+	if skipPreflightChecks {
+		ignoreErrors.Insert("all")
+	}
+	for _, item := range ignoreChecksErrors {
+		lcItem := strings.ToLower(item) // parameters are case insensitive
+		if ignoreErrors.Has(lcItem) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("ignore-checks-errors"), item, "duplicate value"))
```

Comment:
Imo this is unnecessarily strict.  If I say --ignore-checks-errors=Foo,Foo then it is unambiguous that I want Foo skipped :P

In particular I can think of many wrapper-automation scenarios where you want to just append values to this field - and with the above check in place that becomes significantly harder (have to check existing values for duplicates first).

Reply:
That is easy to remove. Original intention was that user must be aware all the time what user tries to ignore. Kind of protection from StackOverflow.

Reply:
+1 to @anguslees